src/pages/packer/plugins/[...page].tsx

```diff
@@ -170,23 +171,30 @@ export async function getStaticProps({ params, ...ctx }) {
 	/**
 	 * Assemble and return static  props for the view
 	 */
-	return {
-		props: {
-			layoutProps: {
-				breadcrumbLinks,
-				githubFileUrl: props.githubFileUrl,
-				headings: props.headings,
-				sidebarNavDataLevels,
-				// Long-form content pages use a narrower main area width
-				mainWidth: 'narrow',
-			},
-			metadata: {
-				title: props.frontMatter.page_title ?? null,
-				description: props.frontMatter.description ?? null,
-			},
-			mdxSource: props.mdxSource,
-			product: productData,
+	const finalProps: DocsViewProps = {
+		layoutProps: {
+			breadcrumbLinks,
+			githubFileUrl: props.githubFileUrl,
+			/**
+			 * TODO: we should likely try to avoid casting here, and instead ensure
```

Comment:
I've tried to fix up `DocsViewProps` related types wherever I can. Usually this is a step towards type safety, but doesn't always get us there. Will be keeping an eye out in the future for opportunities to improve types; I think trying to fix up all related types would balloon the scope of this PR a little too much 🎈

Reply:
Seems fair! I know I've run into the menu items type issues before as well.

Reply:
Yes we've got some `TODO`s building up about this! It's related to the `MenuItem` interface being partially phased out but not fully.

https://github.com/hashicorp/dev-portal/blob/1f3bac6d9a870d0e1f9d7a30e6a5c84af21b97d3/src/views/product-landing/server.ts#L68-L79

We could update the wording in this comment to be more definitive! 😊

Reply:
Sounds good - I added a link in this comment to 🎟 [Clean up references to MenuItem interface](https://app.asana.com/0/1202097197789424/1202405210286689/f) 👍